### PR TITLE
Include Dockerhub in config for circleci in DAGMC repo

### PR DESCRIPTION
**Pull request by NybergWISC**
Add contexts and auth to the different sections. Modeled from https://github.com/pyne/pyne/pull/1344/commits/3ff625db2f7e9988eb1746e67c0f71aed53b0013

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -33,6 +36,9 @@ jobs:
         default: "OFF"
     docker:
       - image: svalinn/dagmc-ci-ubuntu-<< parameters.img >>-<< parameters.compiler >>-ext-hdf5_<< parameters.hdf5 >>-moab_<< parameters.moab >>:latest
+      auth: 
+        username: $DOCKERHUB_USER
+        password: $DOCKERHUB_PASS
```
Comment (ngranda) [inline .circleci/config.yml]: I think it needs one more space. It seems to have problems parsing it

File: .circleci/config.yml [hunk 2]
```diff
@@ -56,7 +62,9 @@ workflows:
   version: 2
   pull_request: # Run only for PullRequest
     jobs:
+      
       - house_keeping:
+          context: Dockerhub
```
Comment (ngranda) [inline .circleci/config.yml]:
```suggestion
          context: dockerhub
```
I think the name of the context in svalinn starts with lowercase `d`

File: .circleci/config.yml [hunk 3]
```diff
@@ -65,6 +73,7 @@ workflows:
               ignore: develop
 
       - build_and_test:
+          context: Dockerhub
```
Comment (ngranda) [inline .circleci/config.yml]:
```suggestion
          context: dockerhub
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -80,6 +89,7 @@ workflows:
   merge:  # Run only on develop change -> when merging a PullRequest
     jobs:
       - build_and_test:
+          context: Dockerhub
```
Comment (ngranda) [inline .circleci/config.yml]:
```suggestion
          context: dockerhub
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -90,6 +100,7 @@ workflows:
             branches:
               only: develop
       - build_and_test:
+          context: Dockerhub
```
Comment (ngranda) [inline .circleci/config.yml]:
```suggestion
          context: dockerhub
```

**Review by ngranda — CHANGES_REQUESTED** [review]
Thanks @NybergWISC. There are a few changes needed to pass CI

**Review by bam241 — APPROVED** [review]
looking good thx !

**Comment by ngranda** [comment]
Thanks @NybergWISC . Merging now

**Pull request merged**